Dynatrace/dynatrace-operator
# Add validation for CR name length

## Description

Please include the following:

It can be a problem when the name of the CR is too long.
This is because we use the name of the CR as a base for other resources we create that are related to it, this is further complicated by the fact that some resource names are used as label values, which have a limit of 63 characters.
- So name-of-CR + suffix + hash-given-by-kubernetes should be shorter then 63 characters

## How can this be tested?
Try to create a CR (dynakube or edgeconnect) with a long name (longer then 40 chars)
(the main thing that should be checked is the message it self 😅 )

## Checklist
